phys2denoise/tests/test_metrics_utils.py

```diff
@@ -20,7 +20,7 @@ def test_mirrorpad_exception(short_arr):
     """When passing array that is too short to perform mirrorpadding, the
     function should give an error."""
     arr = np.array(short_arr)
-    with pytest.raises(Exception) as e_info:
+    with pytest.raises(IndexError):
```

Comment:
I get the expected logger warning here, so the test fails. I'm guessing this one is what you were referring to as the failed catch [here](https://github.com/physiopy/phys2denoise/pull/54#issuecomment-2282224388)? If so, I'm wondering what your suggestion is for handling it to catch the warning instead.

Reply:
Yes that's the one. It's not detected because it's being handled. I'll just add a check that the subsequent warning is raised

Reply:
It's a bit convoluted though and it's pending a full integration of loguru in this package as well. I'd say we merge and document in the issue/PR #53

Reply:
I did a quick fix and it detects the expected warning text (as in the peakdet warning assertion tests I did), however the full integration of loguru needs to be adressed: [Relevant note](https://github.com/physiopy/phys2denoise/issues/48#issuecomment-2308829082)

@m-miedema

Reply:
Thanks @maestroque !